features/discovery/src/test/java/org/opennms/netmgt/discovery/DiscovererBlueprintIT.java

```diff
@@ -0,0 +1,162 @@
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.discovery;
+
+import java.util.Dictionary;
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.activemq.broker.BrokerService;
+import org.apache.camel.test.blueprint.CamelBlueprintTestSupport;
+import org.apache.camel.util.KeyValueHolder;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.opennms.core.test.OpenNMSJUnit4ClassRunner;
+import org.opennms.netmgt.config.DiscoveryConfigFactory;
+import org.opennms.netmgt.config.api.DiscoveryConfigurationFactory;
+import org.opennms.netmgt.config.discovery.DiscoveryConfiguration;
+import org.opennms.netmgt.config.discovery.IncludeRange;
+import org.opennms.netmgt.dao.api.DistPollerDao;
+import org.opennms.netmgt.dao.mock.MockEventIpcManager;
+import org.opennms.netmgt.events.api.EventForwarder;
+import org.opennms.netmgt.events.api.EventIpcManager;
+import org.opennms.netmgt.icmp.Pinger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.test.context.ContextConfiguration;
+
+@RunWith( OpenNMSJUnit4ClassRunner.class )
+@ContextConfiguration( locations = { "classpath:/META-INF/opennms/emptyContext.xml" } )
+public class DiscovererBlueprintIT extends CamelBlueprintTestSupport {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DiscoveryBlueprintIT.class );
+    private static final MockEventIpcManager IPC_MANAGER_INSTANCE = new MockEventIpcManager();
+
+    private static final String LOCATION = "RDU";
+
+    private static BrokerService m_broker = null;
+
+    /**
+     * Use Aries Blueprint synchronous mode to avoid a blueprint deadlock bug.
+     * 
+     * @see https://issues.apache.org/jira/browse/ARIES-1051
+     * @see https://access.redhat.com/site/solutions/640943
+     */
+    @Override
+    public void doPreSetup() throws Exception
+    {
+        System.setProperty( "org.apache.aries.blueprint.synchronous", Boolean.TRUE.toString() );
```

Comment:
Is each test invoked with another JVM fork? If not, we should reset the properties to the default values after the test is done.